plugintypes/execute.go

```diff
@@ -86,13 +86,29 @@ type ExecutePluginOutcome struct {
 	Report cciptypes.ExecutePluginReport `json:"report"`
 }
 
+func (o ExecutePluginOutcome) IsEmpty() bool {
+	return len(o.PendingCommitReports) == 0 && len(o.Report.ChainReports) == 0
+}
+
 func NewExecutePluginOutcome(
 	pendingCommits []ExecutePluginCommitData,
 	report cciptypes.ExecutePluginReport,
 ) ExecutePluginOutcome {
+	pendingCommitsCP := append([]ExecutePluginCommitData{}, pendingCommits...)
+	reportCP := append([]cciptypes.ExecutePluginReportSingleChain{}, report.ChainReports...)
+	sort.Slice(
+		pendingCommitsCP,
+		func(i, j int) bool {
+			return pendingCommitsCP[i].SourceChain < pendingCommitsCP[j].SourceChain
+		})
+	sort.Slice(
+		reportCP,
+		func(i, j int) bool {
+			return reportCP[i].SourceChainSelector < reportCP[j].SourceChainSelector
+		})
```

Comment:
IMO business logic shouldn't be here, move the sorting to plugin.go

Also, should the sequence number be considered when sorting? I forget which data structures are only supposed to have one object per source chain.

Reply:
I'm not sure if this is business logic but encoding seems to have been non-deterministic before, hence this sorting change.

We might want to always ensure this sortation in the `Encode` method rather than depend on the caller / constructor to make sure its sorted.

Reply:
re: sequence number considerations while sorting, are you referring to the case where the same source chain selector can have many different commit reports (in `[]ExecutePluginCommitData`)?

Reply:
Discussed offline, duplicated sorting logic in `Encode`.